src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -87,8 +88,17 @@ private ConcurrentChain CreateChain(int blocksCount)
 
         private async Task WaitUntilQueueIsEmptyAsync()
         {
-            while (this.blockStore.BlocksQueueCount != 0)
-                await Task.Delay(100).ConfigureAwait(false);
+            var queue = this.blockStore.GetMemberValue("blocksQueue") as AsyncQueue<BlockPair>;
+            
+            while (true)
```

Comment:
In case of wrong implementation, you can have infinite loop here. You should put an upper bound here and if it is reached, you should throw exception.

Reply:
Fixed